syntax = "proto3";
package Chat;

enum TYPE{
    controlInfo_C_TO_S = 0;
    updateInfo_S_TO_C = 1;
    setNickName_C_TO_S = 2;
    updateRoomInfo_S_TO_C = 3;
    welcome_S_TO_C = 4;
    notifyStartGame_S_TO_C = 5;
}

message ControlInfo_C_TO_S{
    bool W = 1;
    bool S = 2;
    bool A = 3;
    bool D = 4;
    int32 playerID = 5;
}

// message ControlInfo{
//     bool W = 1;
//     bool S = 2;
//     bool A = 3;
//     bool D = 4;
//     int32 playerID = 5;
// }

message UpdateInfo_S_TO_C{
    int32 frameID = 1;
    repeated ControlInfo_C_TO_S controlInfos = 2;
}

message SetNickName_C_TO_S{
    string nickname = 1;

}

// message NickName_Been_Set_S_TO_C{
// }

message Welcome_S_TO_C{
    int32 playerID = 1;
    int32 kcp_conv = 2;
    string udp_address = 3;
    int32 udp_port = 4;
}

message UpdateRoomInfo_S_TO_C{
    int32 playerCount = 1;
}

message PlayerInfo{
    int32 playerID = 1;
    string nickname = 2;
}

message NotifyStartGame{
    repeated PlayerInfo playerInfos = 1;
}